Libs/vtkSegmentationCore/vtkOrientedImageDataResample.h

```diff
@@ -183,10 +183,25 @@ class vtkSegmentationCore_EXPORT vtkOrientedImageDataResample : public vtkObject
   static bool IsLabelInMask(vtkOrientedImageData* binaryLabelmap, vtkOrientedImageData* mask,
     int extent[6]=nullptr, int maskThreshold=0);
 
+  /// Determine if the segmentation image scalar type is valid for representing segmentations
+  /// \param image Image to validate
+  /// \return True if the image scalar type is valid, false otherwise
+  static bool IsSegmentationScalarTypeValid(vtkImageData* image);
+
+  /// Validate the segmentation image type, casting to another type if necessary.
+  /// \param image Image to validate
+  /// \return True if the image scalar type is valid, or is cast to a valid type, false otherwise
+  static bool ValidateSegmentationImageType(vtkImageData* image);
```

Comment:
This method performs two quite simple operations, which is probably not used that often (and it makes error reporting a bit more complicated). So, I think we probably don't need this.